ring/lifecycler.go

```diff
@@ -102,7 +102,14 @@ func (cfg *LifecyclerConfig) RegisterFlagsWithPrefix(prefix string, f *flag.Flag
 	f.BoolVar(&cfg.EnableInet6, prefix+"enable-inet6", false, "Enable IPv6 support. Required to make use of IP addresses from IPv6 interfaces.")
 }
 
-// Lifecycler is responsible for managing the lifecycle of entries in the ring.
+/*
+Lifecycler is a Service that is responsible for publishing changes to a ring for a single instance.
+
+  - When a Lifecycler first starts, it will be in a [PENDING] state.
+  - After the configured [ring.LifecyclerConfig.JoinAfter] period, it selects some random tokens and enters the [JOINING] state, creating or updating the ring as needed.
+  - The lifecycler will then periodically, based on the [ring.LifecyclerConfig.ObservePeriod], attempt to verify that its tokens have been added to the ring, after which it will transition to the [ACTIVE] state.
```

Comment:
Observe period is a special case that's only useful when using memberlist. Risk we're trying to avoid is that multiple instances join at the same time, and they generate conflicting tokens (this can't happen when using Consul or Etcd thanks to CAS operation). Observe period is used to detect that situation generate more tokens if any conflicting ones were assigned to the other instance.

Reply:
Oh, I'll add a mention of that next time I push another package doc PR.